project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionTests.java

```diff
@@ -268,6 +274,107 @@ public void testProjectSaveWithNoWritePermission() throws IOException {
+
+
+        // Step 1 : Build ProjectB1 and Cache
+        CompileResult depCompileResult = BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/projectB1");
+        if (depCompileResult.getErrorCount() > 0) {
+            Assert.fail("Package B contains compilations error");
+        }
+
+
+        // Step 2 : Build ProjectA with ProjectB as an import
+        Path projectA = RESOURCE_DIRECTORY.resolve("projectA");
+        Project loadProjectA = TestUtils.loadBuildProject(projectA);
+
+
+        // Step 3 : Get compilation of ProjectA and verify dependencies
+        PackageCompilation compilation = loadProjectA.currentPackage().getCompilation();
+        DependencyGraph<ResolvedPackageDependency> dependencyGraph = compilation.getResolution().dependencyGraph();
+        for (ResolvedPackageDependency graphNode : dependencyGraph.getNodes()) {
+            PackageManifest manifest = graphNode.packageInstance().manifest();
+            if (manifest.name().value().equals("projectB")) {
+                Assert.assertEquals(manifest.version().toString(), "1.0.0");
+            }
+        }
+
+
+        // Step 4 : Modify projectA to be blank
+
+        // - Step 4.1 : Get the main.bal file document for ProjectA
+        String newMainProjectAContent = "";
+        Module defaultModuleProjectA = loadProjectA.currentPackage().getDefaultModule();
+        Optional<DocumentId> mainDocumentId =
+                defaultModuleProjectA.documentIds()
+                        .stream()
+                        .filter(documentId -> defaultModuleProjectA.document(documentId).name().equals("main.bal"))
+                        .findFirst();
+        if (mainDocumentId.isEmpty()) {
+            Assert.fail("Failed to retrieve the document ID");
+        }
+
+        // - Step 4.2 : Modify the content of ProjectA
+        Document document = defaultModuleProjectA.document(mainDocumentId.get());
+        document.modify().withContent(newMainProjectAContent).apply();
+
+
+        // Step 5 : Compile ProjectA and verify dependency
+        compilation = loadProjectA.currentPackage().getCompilation();
+        dependencyGraph = compilation.getResolution().dependencyGraph();
+        for (ResolvedPackageDependency graphNode : dependencyGraph.getNodes()) {
+            PackageManifest manifest = graphNode.packageInstance().manifest();
+            if (manifest.name().value().equals("projectB")) {
+                Assert.fail("ProjectB found in dependency after editing ProjectA");
+            }
+        }
+
+
+        // Step 6 : Clear ProjectB cache
+        FileUtils.deleteDirectory(projectBCachePath.toFile());
+
+
+        // Step 7 : Compile ProjectB2 and cache
+        depCompileResult = BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/projectB2");
+        if (depCompileResult.getErrorCount() > 0) {
+            Assert.fail("Package B contains compilations error");
+        }
+
```

Comment:
We should delete the bir cache here. Otherwise the project will be compiled successfully.

Reply:
So do we skip Step 7 alltogether? Because im deleting the cache directory before this, and compiling again so that the new projectB is resolved.

Reply:
After Step 7, you need to repeat Step 6 to be able to compile from the bala that is in the cache

Reply:
Did as you suggested and can verify that the fix works. Removing the fix causes the test to fail.